compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MethodGen.java

```diff
@@ -918,6 +918,10 @@ private void createLocalVariableTable(BIRFunction func, BIRVarToJVMIndexMap inde
         // Add strand variable to LVT
         mv.visitLocalVariable("__strand", String.format("L%s;", STRAND_CLASS), null, methodStartLabel, methodEndLabel,
                               localVarOffset);
+        // Add self to the LVT
+        if (func.receiver != null) {
+            mv.visitLocalVariable("this", String.format("L%s;", B_OBJECT), null, methodStartLabel, methodEndLabel, 0);
```

Comment:
adding "self" instead "this" didn't work?

Reply:
here this is used instead of self because if we use self two local variables get created. One named self and other named this and both get assigned to the same value.

Reply:
I am confused, how is `this` getting added when we add `self`?

Reply:
I think because it is the name of the original variable, while self is a new variable we introduce.

Reply:
hmm wired, I though original name of the variable is `0`

Reply:
It can be seen in the following screenshot:
<img width="792" alt="Screenshot 2021-04-22 at 13 58 00" src="https://user-images.githubusercontent.com/8238078/115696001-e4d0df00-a37f-11eb-917f-1f6fcbf9c18f.png">

Reply:
After further testing now `self` is added to the LVT not `this`